gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegister.java

```diff
@@ -0,0 +1,302 @@
+      throw Throwables.propagate(e);
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  private Table createTable(String dbName, String tableName, Optional<HivePartition> partition, StorageDescriptor sd) {
+    Table table = new Table();
+    table.setParameters(Maps.<String, String> newHashMap());
+    table.getParameters().put("EXTERNAL", Boolean.TRUE.toString());
+    table.setDbName(dbName);
+    table.setTableName(tableName);
+    if (partition.isPresent()) {
+      table.setPartitionKeys(partition.get().getKeys());
+    }
+    table.setOwner(this.hiveOwner);
+    table.setTableType("EXTERNAL_TABLE");
+    table.setSd(sd);
+    return table;
+  }
+
+  /**
+   * Whether any of the given tables exists.
+   */
+  private boolean existsAny(String dbName, Collection<String> tableNames) {
+    for (String tableName : tableNames) {
+      try {
+        if (this.client.tableExists(dbName, tableName)) {
+          return true;
+        }
+      } catch (TException e) {
+        log.error(String.format("Unable to check existence of Hive table %s in db %s", tableName, dbName));
+        throw Throwables.propagate(e);
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Whether any of the given partitions exists.
+   */
+  private boolean existsAny(String dbName, String tableName, Collection<HivePartition> partitions) {
+    for (HivePartition partition : partitions) {
+      try {
+        client.getPartition(dbName, tableName, partition.getValues());
+        return true;
+      } catch (NoSuchObjectException e) {
+        // Partition does not exist. Nothing to do
+      } catch (TException e) {
+        log.error(String.format("Unable to check existence of Hive partition %s in table %s db %s", partition,
+            tableName, dbName));
+        throw Throwables.propagate(e);
+      }
+    }
+    return false;
+  }
+
+  private void dropObsoleteTables(String dbName, Collection<String> tableNames) throws TException {
+    for (String tableName : tableNames) {
+      Lock lock = locks.get(dbName + tableName);
+      lock.lock();
+
+      try {
+        if (this.client.tableExists(dbName, tableName)) {
+          this.client.dropTable(dbName, tableName);
+        }
+      } finally {
+        lock.unlock();
+      }
+    }
+  }
+
+  private void dropObsoletePartitions(String dbName, String tableName, Collection<HivePartition> partitions)
+      throws TException {
+    Lock lock = locks.get(dbName + tableName);
+    lock.lock();
+
+    try {
+      for (HivePartition partition : partitions) {
+        try {
+          client.dropPartition(dbName, tableName, partition.getValues(), false);
+        } catch (NoSuchObjectException e) {
+          // Partition does not exist. Nothing to do
+        }
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  private void addPartition(Table table, HivePartition partition, StorageDescriptor sd) throws TException {
```

Comment:
By default, the old code based would register the Avro schema using `avro.schema.literal` (the value would be the actual schema), but only if the schema did not exceed a certain size. If it exceeded a size it would register the schema using `avro.schema.url` and the value would be a URI pointing to a file (typically on HDFS), that contains the schema.

The reason this was put in is that Hive would choke of the `avro.schema.literal` was too long, and the table would be un-querable (I believe Identity.Profile hit this issue); which is why this feature was added.

However, that was a while ago, so I'm not sure if that bug is still in Hive.